core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackRuntimeVersionValidator.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.controller.validation.stack;
+
+import static com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.StackV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.cluster.ClusterV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.cluster.cm.ClouderaManagerV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.cluster.repository.RepositoryV4Request;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.service.datalake.SdxClientService;
+import com.sequenceiq.sdx.api.model.SdxClusterResponse;
+
+@Component
+public class StackRuntimeVersionValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackRuntimeVersionValidator.class);
+
+    @Inject
+    private SdxClientService sdxClientService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public void validate(StackV4Request stackRequest) {
+        if (isDifferentDataHubAndDataLakeVersionAllowed()) {
+            LOGGER.debug("The Data Hub version validation has been turned off with entitlement.");
+        } else {
+            LOGGER.debug("Validating Data Hub version.");
+            findStackVersion(stackRequest).ifPresent(stackRuntimeVersion -> {
```

Comment:
I think we might need to check the image as well. In the case of pre-warm image, the request won't contain the repository info, but the image id. In this case, we should check the version on the image if this is available at this point.